compaction/compaction_manager.cc

```diff
@@ -934,11 +934,9 @@ future<> compaction_manager::stop() {
     if (auto cm = std::exchange(_task_manager_module, nullptr)) {
         co_await cm->stop();
     }
-    // never started
-    if (_state == state::none) {
-        co_return;
-    } else {
-        do_stop();
+
+    do_stop();
+    if (_state != state::none) {
```

Comment:
This may be safe to do but the commit message implies this patch doesn't change functionality, but it seems to do.